File: backend/src/main/java/br/ufpr/sept/so2/modules/iam/domain/SenhaReutilizadaException.java
package br.ufpr.sept.so2.modules.iam.domain;

import br.ufpr.sept.so2.shared.domain.exception.DadoInvalidoException;

public class SenhaReutilizadaException extends DadoInvalidoException {

    public SenhaReutilizadaException(String message) {
        super(message);
    }

    public static SenhaReutilizadaException historico() {
        return new SenhaReutilizadaException(
                "Esta senha já foi utilizada recentemente. Escolha uma senha diferente.");
    }

    public static SenhaReutilizadaException temporaria() {
        return new SenhaReutilizadaException(
                "A nova senha não pode ser igual à senha temporária gerada pelo sistema.");
    }
}

File: backend/src/main/kotlin/br/ufpr/sept/so2/modules/iam/domain/SenhaReutilizadaException.kt
package br.ufpr.sept.so2.modules.iam.domain

import br.ufpr.sept.so2.shared.domain.exception.DadoInvalidoException

class SenhaReutilizadaException(message: String) : DadoInvalidoException(message) {
    companion object {
        fun historico(): SenhaReutilizadaException =
            SenhaReutilizadaException(
                "Esta senha já foi utilizada recentemente. Escolha uma senha diferente.",
            )

        fun temporaria(): SenhaReutilizadaException =
            SenhaReutilizadaException(
                "A nova senha não pode ser igual à senha temporária gerada pelo sistema.",
            )
    }
}
